core/arch/arch.c

```diff
@@ -739,10 +739,12 @@ d_r_arch_init(void)
                   offsetof(priv_mcontext_t, simd)));
     ASSERT(offsetof(app_state_at_intercept_t, mc) ==
            offsetof(app_state_at_intercept_t, start_pc) + sizeof(void *));
+#ifdef X86
```

Comment:
The assert below does not look x86-specific at all: `unprotected_context_t` has some misc fields and then CLEANCALL_NUM_INLINE_SLOTS reg_t fields, on all platforms (only ifdef is `dr_errno` covered by the IF_WINDOWS).  This could be improved by `s/5/CLEANCALL_NUM_INLINE_SLOTS/`.  But I don't think this x86 define makes sense here.

Reply:
How about getting rid of the assert altogether? I don't see the point of it, really. It just seems to restating, incorrectly, how a C compiler computes the size of a struct. It's incorrect because it fails to take account of alignment.

Soon, perhaps later today, I will be suggesting a much better way of checking that the offsets in `.asm` files are correct.

Reply:
I assume, as the comment says, it is checking asm offsets, so removing such a check doesn't seem good.  x86.asm does have things like `UPCXT_BEFORE_INLINE_SLOTS` it appears which are relevant: used in `dstack_OFFSET` looks like.  It looks like aarch64.asm also has `dstack_OFFSET` directly hardcoded: but that value changing by an `unprotectext_context_t` change would also be caught by this assert, which is nicer than a weird crash.  So doesn't it seem like this assert is providing a worthwhile service on multiple platforms?

Reply:
Here's my suggestion for how to check offsets at compile time: #7202 

It's an early draft. I was going to add a few more macros besides `dstack_OFFSET` before submitting it, but perhaps it makes more sense to get your opinion first. There are lots of alternatives I've considered but this way looks like a good compromise between simplicity and helpfulness. Something like this would have saved me time working on this patch.

Reply:
I've added an `ALIGN_FORWARD` into the `ASSERT`.